packages/ember-glimmer/tests/integration/syntax/with-test.js

```diff
@@ -30,4 +38,246 @@ moduleFor('Syntax test: {{#with}}', class extends SharedConditionalsTest {
 
     this.assertText('False');
   }
+
+  ['@test can access alias and original scope']() {
+    this.render(`{{#with person as |tom|}}{{title}}: {{tom.name}}{{/with}}`, {
+      title: 'Señor Engineer',
+      person: { name: 'Tom Dale' }
+    });
+
+    this.assertText('Señor Engineer: Tom Dale');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('Señor Engineer: Tom Dale');
+
+    this.runTask(() => {
+      set(this.context, 'person.name', 'Yehuda Katz');
+      set(this.context, 'title', 'Principal Engineer');
+    });
+
+    this.assertText('Principal Engineer: Yehuda Katz');
+
+    this.runTask(() => {
+      set(this.context, 'person', { name: 'Tom Dale' });
+      set(this.context, 'title', 'Señor Engineer');
+    });
+
+    this.assertText('Señor Engineer: Tom Dale');
+  }
```

Comment:
This also tests "#with without inverse" implicitly – which the `templateFor` suite does not cover